From 9ae13c9b829cd4fc544b8ee0d138d9e78d171549 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 14 Sep 2020 02:40:17 +0000
Subject: [PATCH] libmp3lame/vector/Makefile.am: Add msse to fix i686
 compilation

../../../libmp3lame/vector/xmm_quantize_sub.c:72:34: error: always_inline function '_mm_loadu_ps' requires target feature 'sse', but would be inlined into function 'init_xrpow_core_sse' that is compiled without support for 'sse'
    const __m128 vec_fabs_mask = _mm_loadu_ps(&fabs_mask._float[0]);
                                 ^
../../../libmp3lame/vector/xmm_quantize_sub.c:80:27: error: always_inline function '_mm_set_ps1' requires target feature 'sse', but would be inlined into function 'init_xrpow_core_sse' that is compiled without support for 'sse'
    vec_xrpow_max._m128 = _mm_set_ps1(0);
                          ^

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 libmp3lame/vector/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libmp3lame/vector/Makefile.am b/libmp3lame/vector/Makefile.am
index fa0b6677..c6c798bc 100644
--- a/libmp3lame/vector/Makefile.am
+++ b/libmp3lame/vector/Makefile.am
@@ -6,7 +6,7 @@ if WITH_XMM
 noinst_LTLIBRARIES = liblamevectorroutines.la
 endif
 
-##liblamecpuroutines_la_LIBADD = 
+##liblamecpuroutines_la_LIBADD =
 ##liblamecpuroutines_la_LDFLAGS =
 
 INCLUDES = @INCLUDES@ \
@@ -20,6 +20,7 @@ xmm_sources = xmm_quantize_sub.c
 
 if WITH_XMM
 liblamevectorroutines_la_SOURCES = $(xmm_sources)
+liblamevectorroutines_la_CFLAGS = -msse
 endif
 
 noinst_HEADERS = lame_intrin.h
-- 
2.25.1

